### ISSUE TITLE
The submission file dialog and the inspect page mess up lines and line numbers when there are over nine lines

### ISSUE THREAD

**Issue by Mankro** [apluslms/a-plus#794]
When you open a submission, either in the feedback or the teacher's inspect submission page, there is the button for opening the submitted file. It shows the file contents in a modal dialog. This is especially useful for source code files and it also highlights the code syntax automatically. However, if the file has over nine lines, the line numbers can get messed up so that they do not fit into a single line and consume too much vertical space. The whole line gets then messed up.

This screenshot shows the problem. Some of the code is censored because it is a student's submission.

![screenshot-student-censored](https://user-images.githubusercontent.com/6130378/111496960-63b37600-8749-11eb-83df-f203c770d9bd.png)

**This problem does not occur in all browsers. Or in all submissions. Actually, the same browser has the problem with some submissions while other submissions work correctly.**

My example that actually works correctly in this screenshot:
![screenshot-file-modal-good](https://user-images.githubusercontent.com/6130378/111497411-c1e05900-8749-11eb-97c2-090961857e31.png)

https://plus.cs.aalto.fi/aplus-manual/master/programming_exercises/hello_world/programming_exercises_hello_world_python/submissions/3818778/inspect/

The file dialog uses some custom JavaScript code for setting up the dialog:
https://github.com/apluslms/a-plus/blob/ae4aef150899ad3be6402fd37c90345b6713d8c2/assets/js/aplus.js#L184
https://github.com/apluslms/a-plus/blob/ae4aef150899ad3be6402fd37c90345b6713d8c2/assets/js/aplus.js#L365

**Comment by Mankro** [comment]
This is not only related to the web browser used. The same user using the same browser can see this buggy behaviour in some student submissions while others work correctly. This indicates that the A+ JavaScript code could be buggy for preparing the file dialog.

The teacher who reported this uses Mac and Firefox. I tested this in Linux in both Firefox and Chrome. The same browser works correctly in some submissions while other submissions do not work!

Internal ticket with links to submissions: https://rt.cs.aalto.fi/Ticket/Display.html?id=18516

**Comment by Mankro** [comment]
Now that investigate this again, the same submissions seem to work correctly. We are not able to reproduce the error cases. How could this happen?

**Comment by Mankro** [comment]
New report in A+ v1.11 and the new inspect page:
https://github.com/apluslms/a-plus/issues/936#issue-1079481875
https://github.com/apluslms/a-plus/issues/936#issuecomment-993330121

Decreasing the zoom level in the browser may fix this as a workaround. There must be something wrong in our CSS styles, though. Maybe the padding, width or margins.

**Comment by ihalaij1** [comment]
I noticed that this issue only happened in exercises that use python-grader-utils.

I identified the cause of the problem in python-grader-utils' CSS, where too broad of a selector for whitespace pre-wrap was used to disable horizontal scrolling of preformatted text in the feedback.

I don't know how recently the line-numbering bug has started appearing, but the CSS selector has existed for a long time. Some other updates to the styles may have caused it to become a noticeable issue.

I have pushed a fix in python-grader-utils repository, so this issue can be closed:
https://github.com/apluslms/python-grader-utils/commit/2e0d2c695d21ce1fff53943a255196a23840febb
